### ISSUE TITLE
Case of zero collateral transaction is not covered by specification and implementation

### ISSUE THREAD

**Issue by uhbif19** [IntersectMBO/formal-ledger-specifications#78]
Babbage specification requires `collInputs tx ̸= ∅` in `feesOK`. While in case when fee (and consequently collateral amount) is zero `collInputs` may be empty.

We need that case while working with [Hydra](https://hydra.family/). Collaterals seems to be the only reason to commit ADA onto L2 for us.

Details of our case are in this issue: https://github.com/mlabs-haskell/hydra-auction/issues/193

**Comment by WhatisRT** [comment]
We could make this conditional on the `Network` the ledger runs on or a global constant more generally.

If you want this any time soon, you should make an issue at https://github.com/input-output-hk/cardano-ledger, since we're mostly occupied with the next era & don't even have any Plutus logic in here at all. It's also not entirely clear what you want: do you just want a change in the spec, or do you want a change in the implementation as well?

**Comment by WhatisRT** [comment]
Yes, there are a bunch of reasons that maybe don't apply to Hydra. It acts as DOS protection, and it ensures that the same transaction ID can't appear twice on the chain.

**Comment by WhatisRT** [comment]
This issue got buried, and I don't think we're going to do anything about it. Feel free to reopen if necessary.

To answer the previous question, if you have a failing script without any collateral input none of the regular inputs get spent. So if that's a valid transaction you can post it on chain as many times as you want. It has zero effect on the state, so maybe that's fine, but it would be an exception to the rule that you cannot have multiple identical transactions.

I'm not sure what exactly the requirements of Hydra are, but I think deviating from the regular logic (and ideally clearly documenting it somewhere) seems like a good choice.
